src/triage/experiments/base.py

```diff
@@ -595,6 +604,17 @@ def _run(self):
 
         self.train_and_test_models()
         logging.info("Experiment complete")
+        if self.profile:
+            cp.disable()
```

Comment:
Bikeshedding, but which you _might_ enjoy.

* Rather than tracking this stuff within `_run`, you could ask the profiler to execute `_run`; _e.g._, in `run`:

```
def _run_profile(self):
    cp = cProfile.Profile()
    cp.runcall(self._run)
    store = …
    …

def run(self):
    try:
        if self.profile:
            self._run_profile()
        else:
            self._run()
    except …
        …
```

* If it's important to keep this all in `_run`, it might be useful for variable management to essentially eagerly translate the `profile` attribute into the `Profile` object. And, related, I think it's important to ensure that profile is, finally, disabled. _E.g._:

```
cp = cProfile.Profile() if self.profile else None
if cp:
    cp.enable()

try:
    try:
        …
    …
finally:
    cp.disable()
…
if cp:
    store = …
…
```